axelor-sale/src/main/java/com/axelor/apps/sale/service/SaleOrderGeneratorServiceImpl.java

```diff
@@ -94,4 +97,16 @@ protected void checkClientPartner(Partner clientPartner, SaleOrder saleOrder)
           company.getName());
     }
   }
+
+  protected void checkinAti(SaleOrder saleOrder, String inAti) throws AxelorException {
+    Boolean ati = Boolean.parseBoolean(inAti);
+    Company company = saleOrder.getCompany();
+    if (company.getSaleConfig().getSaleOrderInAtiSelect() == 1
+        || company.getSaleConfig().getSaleOrderInAtiSelect() == 2) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_INCONSISTENCY,
+          I18n.get(SaleExceptionMessage.ATI_CHANGE_NOT_ALLOWED));
+    }
+    saleOrder.setInAti(ati);
```

Comment:
Do not set the inAti in the check method. Create another method that check if inAti is not null then you can set it.